### DISCUSSION TITLE
How can you run the migration helper to v10 in CLI mode?

### DISCUSSION THREAD

**Discussion by queicherius** [cypress-io/cypress#22939] [Questions and Help] [answered]
With Cypress 10 and all the changes required to upgrade, you introduced a migration helper in GUI mode.

I can not find any information about how to run this headless/without the UI. Is there one currently / is there one planned? I am always running Cypress on the CLI only, on a server without a display connected, and this would be super helpful. Thanks!

**Comment by lmiller1990** [comment]
Currently there is no CLI - you need to use the UI. There's a few decisions you need to make, and the UI makes it more clear what's going on.

Generally, I'd recommend opening Cypress 10 at least once locally and completing the migration process via the UI.

**Comment by edaldridge** [comment]
Using a GUI is not possible on a server that doesn't have one, I use Jenkins to run tests on a headless Linux server.  At the moment I am having to run all my tests on 9.7 and maintain two folders in my repo, one for locally using version 10 and the other to allow Jenkins to work.

Would be great if there was a step by step guide to upgrading via the CLI as I can't believe I am the only one in this position

**Reply by lmiller1990** [reply]
Would it be possible to just run Cypress 10 locally once, update, commit and push?

You can definitely upgrade manually. There is some info here https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-version-10-0

If you can share your `cypress.json` and the contents of your `cypress` directory (not the specs, but everything else, like `plugins` etc) I could probably give it a try for you. The main changes are 

- cypress.json -> cypress.config.js 
- pluginsFile -> import to `setupNodeEvents` in `cypress.config.js`

We also recommend changing the extension to use `.cy.js` to not conflict with other test runners (like Jest) but if you want to do the minimal changes possible, I can help with that, once I can see your current cypress configuration.

**Reply by edaldridge** [reply]
Thanks for the reply Lachlan! 

I'll have a read through the docs you've provided and see if I can work out the steps to take.  It is probably something pretty simple but the first time Jenkins tried to load 10 it caused so many issues I've been really nervous about changing it again.

**Reply by lmiller1990** [reply]
It might seem like a lot, but it should take <1m if you use the migration UI we built.

If you really don't want to, you could share a minimal version of you project, and I could guide you -- but unless there's a major blocker for running the UI locally (with npx cypress open, just once) - walking through the migration, and committing, that would be the easiest way.

**Reply by edaldridge** [reply]
It's not a question of not wanting to run the UI, the server that these tests are running on has no GUI component installed so there is no way I _can_  run the UI.  

I think what I need to know is what files the UI install modifies and if it does anything other than change the name of a file and change the location of the tests because if it's just a case of renaming a couple of files and changing where the tests point to then it's fairly easy but if the migration wizard does stuff under the hood then it's going to be a bit more of a hassle to get this done.

**Reply by lmiller1990** [reply]
Right, I see.

I think you can do this entire migration only changing one file, which is `cypress.json` -> `cypress.config.js`. The main fields that have changed are `integrationFolder` and `testFiles` got merged into a single property: `specPattern`, and `pluginsFile` is now `setupNodeEvents`. An example can be found in our tests: [before](https://github.com/cypress-io/cypress/blob/develop/system-tests/projects/migration-e2e-custom-integration-default-value-and-testFiles/cypress.json) and [after](https://github.com/cypress-io/cypress/blob/develop/system-tests/projects/migration-e2e-custom-integration-default-value-and-testFiles/expected-cypress.config.js). There is a bunch more examples - anything starting with `migration-e2e` in [this directory](https://github.com/cypress-io/cypress/tree/develop/system-tests/projects) contains a `cypress.json` and an `expected-cypress.config.js`.

Alternatively: If you just want to migrate and move on with more important things, I can just do it for you, since I know all the changes.

Can you share your `cypress.json` file here? It'll be faster if I just migrate it by hand, I can't promise it'll work perfectly first try, but I can get you 90% of the way there.

**Reply by edaldridge** [reply]
`{
    "projectId": "9dwiye",
    "chromeWebSecurity": false,
    "defaultCommandTimeout": 50000,
    "experimentalFetchPolyfill": true,
    "viewportWidth": 1920,
    "viewportHeight": 1080,
    "trashAssetsBeforeRuns": true
}`

**Reply by lmiller1990** [reply]
Nice, mostly defaults. Give this a try. We have pretty good error handling; if something isn't quite correct, it should print a useful message. I'm not sure if your project uses TypeScript or any plugins, so I just did the most minimal, basic changes. 

Let me know how it goes.

```js
module.exports = {
  projectId: "9dwiye",
  chromeWebSecurity: false,
  defaultCommandTimeout: 50000,
  experimentalFetchPolyfill: true,
  viewportWidth: 1920,
  viewportHeight: 1080,
  trashAssetsBeforeRuns: true,
  e2e: {
    supportFile: "cypress/support/index.js",
    specPattern: "cypress/integration/**/*",
    setupNodeEvents(on, config) {
      return require("./cypress/plugins/index.js")(on, config);
    },
  },
};
```

**Reply by edaldridge** [reply]
Thank you for the help Lachlan, I may not get a chance to try this for a while but I wanted to let you know I'd seen it and will post back and let you know how it goes!

**Reply by lmiller1990** [reply]
No worries, let me know how it goes.

**Comment by edaldridge** [comment]
Hi Lachlan

My apologies for not coming back on this sooner.. I’d been putting it off but the recent update to 12 has forced me to finally start looking at this.


I’ve uploaded the cypress.config.js file with the entries you gave me, I am getting an error below :



“Your configFile is invalid: /opt/jenkins/workspace/HaywardsHeath/DDMS/aftJapan/QA Server/atfJapanAddCustomer/cypress.config.js

It threw an error when required, check the stack trace below:

Error: Cannot find module 'cypress'” .



Jenkins installs Cypress through a Node integration with the -g flag, I can’t see anything wrong with that because it’s always worked with v9 of Cypress.  Was hoping you had a bright idea I could try?



Thanks
Ed

**Comment by edaldridge** [comment, marked as answer]
Never mind, I managed to resolve the issue.  Peace and tranquillity reign again

**Reply by lmiller1990** [reply]
May peace and tranquillity reign again continue forevermore, we will not make any major changes like this for the foreseeable future.
